src/ai/backend/storage/api/vfolder/types.py

```diff
@@ -0,0 +1,130 @@
+import uuid
+from pathlib import Path, PurePath, PurePosixPath
+from typing import List, Optional, TypeAlias
+
+from pydantic import AliasChoices, Field
+from pydantic import BaseModel as PydanticBaseModel
+
+from ai.backend.common.types import BinarySize, QuotaConfig, QuotaScopeID, VFolderID
+
+
+class BaseModel(PydanticBaseModel):
+    """Base model for all models in this module"""
+
+    model_config = {"arbitrary_types_allowed": True}
+
+
+VolumeID: TypeAlias = uuid.UUID
+
+
+# Common fields for VolumeID and VFolderID
+VOLUME_ID_FIELD = Field(
+    ...,
+    validation_alias=AliasChoices(
+        "volume",
+        "volumeid",
+        "volume_id",
+        "volumeId",
+    ),  # Previous name was `v`
+    description="A unique identifier for the volume.",
+)
+VFOLDER_ID_FIELD = Field(
+    ...,
+    validation_alias=AliasChoices(
+        "vfid",
+        "vfolderid",
+        "vfolder_id",
+        "vfolderId",
+    ),
+    description="A unique identifier for the virtual folder.",
+)
+QUOTA_SCOPE_ID_FIELD = Field(
+    ...,
+    validation_alias=AliasChoices(
+        "qsid",
+        "quotascopeid",
+        "quota_scope_id",
+        "quotaScopeId",
+    ),
+    description="A unique identifier for the quota scope.",
+)
+
+
+class VolumeIDModel(BaseModel):
+    volume_id: VolumeID = VOLUME_ID_FIELD
+
+
+class VolumeMetadataModel(BaseModel):
+    """For `get_volume`, `get_volumes`"""
+
+    volume_id: VolumeID = Field(..., description="The unique identifier for the volume.")
+    backend: str = Field(
+        ..., description="The backend storage type for the volume (e.g., CephFS, GPFS)."
+    )
+    path: Path = Field(..., description="The path where the volume is mounted.")
+    fsprefix: Optional[PurePath] = Field(
+        default=None, description="The filesystem prefix for the volume, or None if not applicable."
+    )
+    capabilities: list[str] = Field(
+        ..., description="A list of capabilities supported by the volume."
+    )
+
+
+class VolumeMetadataListModel(BaseModel):
+    volumes: List[VolumeMetadataModel] = Field(..., description="A list of volume information.")
+
+
+class VFolderIDModel(BaseModel):
+    volume_id: VolumeID = VOLUME_ID_FIELD
+    vfolder_id: VFolderID = VFOLDER_ID_FIELD
+    # For `get_vfolder_info`: mount
+    subpath: Optional[PurePosixPath] = Field(
+        default=None,
+        description="For `get_vfolder_info`\n\
+            The subpath inside the virtual folder to be queried.",
+    )
+    # For `clone_vfolder`
+    # You can use volume_id and vfolder_id as src_volume and src_vfolder_id.
+    dst_vfolder_id: Optional[VFolderID] = Field(
+        default=None,
+        validation_alias=AliasChoices(
+            "dst_vfid",
+            "dstvfolderid",
+            "dst_vfolder_id",
+            "dstVfolderId",
+        ),
+        description="For `clone_vfolder`\n\
+            The destination virtual folder ID to clone to.",
+    )
```

Comment:
I guess the models in this module are used to serialize and validate input of API handlers.
This `VFolderIDModel` seems to be used in too many cases.
Let's divide this into separate models.